src/platform/Zephyr/ConfigurationManagerImpl.cpp

```diff
@@ -220,14 +225,20 @@ void ConfigurationManagerImpl::DoFactoryReset(intptr_t arg)
     }
 #else
 
+    ConnectivityMgr().ErasePersistentInfo();
+    //! Workaround for NCS issue #19795
```

Comment:
Why this is workaround, isn't just optimization. The higher erase time with MPSL is justified, by disabling MPSL we do have direct access to flash, thus it's faster. What do you think?

Reply:
Hmm, ok that's right, I was thinking about that as a workaround for the time penalty that mpsl causes. But ok, in this context it is optimization. :)

Reply:
I've removed "workaround".